test/zone/config_test.rb

```diff
@@ -32,4 +32,12 @@ def test_config_is_properly_loaded_from_zonefile
     assert_equal [{type: 'NS'}, {type: 'A', fqdn: 'a.example.com.'}], zone.config.ignore_patterns
     assert_equal 'DynECT', zone.config.provider
   end
+
+  def test_config_supports_alias_based_on_provider
+    config = Zone.new('dynect-config.com', config: {provider: 'DynECT'}).config
+    refute_predicate config, :supports_alias?
+
+    config = Zone.new('dnsimple-config.com', config: {provider: 'DNSimple'}).config
+    assert_predicate config, :supports_alias?
+  end
```

Comment:
@andrewlouis93 

emil is suggesting we add config like this... https://github.com/Shopify/record_store/blob/34a02f9f6d5972039c8dd0b8e165fed31fdb4bf8/test/zone/config_test.rb#L25

and test for the value being set similar to this...
https://github.com/Shopify/record_store/blob/34a02f9f6d5972039c8dd0b8e165fed31fdb4bf8/test/zone/config_test.rb#L33